src/DocumentPicker.Plugin/Shared/DocumentConverterService.cs

```diff
@@ -98,24 +99,65 @@ public async Task<DocumentConverterResult> ConvertPdfToSvgStringAsync(string fil
             }
         }
 
-        public void Validate(DocumentConverterResult result)
+        public async Task ConvertPdfToSvgAsync(Stream inputStream, Stream outputStream, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                using (var doc = await PdfDocument.OpenAsync(inputStream, cancellationToken: cancellationToken))
+                {
+                    if (doc.Pages.Count > 1)
+                        throw new DocumentConverterException("Pdf document contains multiple pages - provide a single-page document");
+
+                    using (var memoryStream = new MemoryStream())
+                    {
+                        await doc.Pages[0].SaveAsSvgAsync(memoryStream, cancellationToken: cancellationToken);
+
+                        var buffer = memoryStream.GetBuffer();
+
+                        memoryStream.Seek(0, SeekOrigin.Begin);
+                        if (HasOnlyImages(memoryStream))
+                            throw new DocumentConverterException(
+                                "The PDF page seems to be from a scanned document. Please upload this plan as image instead (.jpeg, .png)");
+
+                        await outputStream.WriteAsync(buffer, 0, (int)memoryStream.Length, cancellationToken);
```

Comment:
waaat? 😅

please add  another 
`memoryStream.Seek(0, SeekOrigin.Begin);` to line 121
and in line 122, simply call
`await memoryStream.CopyToAsync(outputStream, cancellationToken);` 

https://learn.microsoft.com/en-us/dotnet/api/system.io.stream.copytoasync?view=net-6.0


Currently, you are copying from the output stream to the memory stream - this does not makes sense, does it? ;-)

Reply:
No first i save the buffer and then write it in the outputStream. Did not use CopyTo sorry